integ/scripts/bash/rfdk-integ-e2e.sh

```diff
@@ -90,11 +91,35 @@ for COMPONENT in **/cdk.json; do
     export ${COMPONENT_NAME}_START_TIME=$SECONDS
     if [[ "$COMPONENT_NAME" != _* ]]; then
         # Excecute the e2e test in the component's scripts directory
-        cd "$INTEG_ROOT/$COMPONENT_ROOT" && ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" || cleanup_on_failure
+        cd "$INTEG_ROOT/$COMPONENT_ROOT"
+        if [ "${RUN_TESTS_IN_PARALLEL}" = true ]; then
+            ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" || cleanup_on_failure &
```

Comment:
Yeah this interleaved output was a problem I was afraid of and had suggested maybe each test suite could write to its own file, but that gets complicated when we are running in CodeBuild. I don't know how easy it would be to write those to CloudWatch.

Reply:
Yeah, at the very least, we could write the output to files and the `cat` them after all tests have complete if there were any deployment failures. Ideally though, we'd also have some form of status being output as the tests are running to be able to observe the progress of each component test running (deploying/testing/destroying/complete/failed).


```sh
$ yarn run e2e
2021-02-03 16:39:00 [DL] deploying...
2021-02-03 16:39:00 [RQ] deploying...
2021-02-03 16:39:00 [WF] deploying...
2021-02-03 16:39:00 [WFS] deploying...
2021-02-03 17:04:00 [DL] testing...
2021-02-03 17:21:00 [RQ] testing...
2021-02-03 20:00:00 [WF] testing...
2021-02-03 20:03:00 [DL] destroying...
...
```

If a deployment failed for one of those components, we could output the deployment log file after all tests are completed (so it isn't interleaved with the above output).

What do you think?

Reply:
I like it. Much simpler than having to work with CloudWatch.